app/controllers/concerns/recaptcha_concern.rb

```diff
@@ -0,0 +1,30 @@
+module RecaptchaConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :allow_csp_recaptcha_src, if: :recaptcha_enabled?
+  end
+
+  private
+
+  RECAPTCHA_SCRIPT_SRC = [
+    'https://www.google.com/recaptcha/',
+    'https://www.gstatic.com/recaptcha/',
+  ].freeze
+
+  RECAPTCHA_FRAME_SRC = [
+    'https://www.google.com/recaptcha/',
+    'https://recaptcha.google.com/recaptcha/',
+  ].freeze
+
+  def allow_csp_recaptcha_src
+    policy = current_content_security_policy
+    policy.script_src(*policy.script_src, *RECAPTCHA_SCRIPT_SRC)
+    policy.frame_src(*policy.frame_src, *RECAPTCHA_FRAME_SRC)
+    request.content_security_policy = policy
+  end
+
+  def recaptcha_enabled?
+    false
```

Comment:
Just for my understanding, this is always false because it will be overridden in the controllers where the concern is included?

Reply:
Yeah, it's another case of possible premature abstraction, where the feature flag we're introducing here is specific to phone setup, but I didn't want to make the otherwise-generic reCAPTCHA concern aware of that feature flag. Instead, it's expected to be overridden by the controller, which _would_ have awareness of the specific feature flag.

Reply:
Also, the idea with choosing `false` as the default is that it'd be pretty obviously broken if someone didn't override it.

Reply:
Yep, makes sense!